src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -314,11 +279,9 @@ public void clusterChanged(ClusterChangedEvent event) {
                                     //  - we acquired all shards locks here --> we can import the dangling index
                                     //  - we failed to acquire the lock --> somebody else uses it - DON'T IMPORT
                                     //  - we acquired successfully but the lock list is empty --> no shards present - DON'T IMPORT
-                                    // in the last case we should in-fact try to delete the directory since it might be a leftover...
+                                    // in the last case be careful and just do nothing
```

Comment:
now that we delete the state, do we still need to do this? do you see any case where the state file deletion can fail? also, given the fact that the state file can be on master nodes (where there are no shard locks) I'm not sure this adds something now, right?